ignite/metrics/metric.py

```diff
@@ -191,12 +194,14 @@ def compute(self):
     """
 
     # public class attribute
-    required_output_keys = ("y_pred", "y")
+    required_output_keys = ("y_pred", "y")  # type: Optional[Tuple]
     # for backward compatibility
     _required_output_keys = required_output_keys
 
     def __init__(
-        self, output_transform: Callable = lambda x: x, device: Union[str, torch.device] = torch.device("cpu"),
+        self,
+        output_transform: Optional[Callable] = lambda x: x,
```

Comment:
Maybe, better to have 
```suggestion
        output_transform: Callable = lambda x: x,
```
?

Reply:
The reason behind the `Optional` here is because of the `RunningAverage` class, where the `output_transform` can be None
https://github.com/pytorch/ignite/blob/bcac5c52f5a54aca9f874ab663ed217cfe107a65/ignite/metrics/running_average.py#L49-L56

Reply:
That's true :) Oh, what a badly designed library :)

Reply:
I think the best we can do is to put ignores what those bad designs inside Frequency.attach and RunningAverage.__init__ and create an issue to fix them. And we annotate correctly the args where we intend the correct behaviour like `Metric.__init__(output_transform: Callable = lambda x: x)`

Reply:
Thanks a lot for checking all that @gruebel !

Reply:
Sure thing 💪 for me it is just hard to decide, if the code is intended to be like this or it is really a bug.